Updated DatadogOrganizationProperties for reduce onboarding friction story | Datadog

harsh007kumar opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

harsh007kumar committed:
initial commit

harsh007kumar committed:
updated new api version 2022-06-01 in all examples and datadog.json

harsh007kumar committed:
updated 2 parameters of 'DatadogOrganizationProperties' to mandatory:
- Name : Name of the Datadog organization
- Id : Id of the Datadog organization

harsh007kumar committed:
added datadogOrgID in 'datadogOrganizationProperties'
In Request send for Create Monitor

harsh007kumar committed:
added New Tag version details in all readme(s) wherever required

harsh007kumar committed:
Change Mutability for DatadogOrganizationProperties => Id & Name (added read to x-ms-mutability)

harsh007kumar commented:
Most of the changes are already reviewed and merged to private repo, check => https://github.com/Azure/azure-rest-api-specs-pr/pull/7482 for details.

raising this PR to Generate SDKs/update CLI with new api version.

dw511214992 commented:
> ### Generated ApiView
> Language	Package Name	ApiView Link
> Python	track2_azure-mgmt-datadog	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and re-trigger the CI. If issue still exists, please ask PR assignee for help
> Java	azure-resourcemanager-datadog	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and re-trigger the CI. If issue still exists, please ask PR assignee for help
> JavaScript	@azure/arm-datadog	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and re-trigger the CI. If issue still exists, please ask PR assignee for help

If you don't want to get apiview, please ignore this comment. thanks

harsh007kumar commented:
@ArcturusZhang I see in new API Version 3 checks are failing 2 of them because
I have updated the property of 'DatadogOrganizationProperties' ID & Name to remove readonly attribute and added
"x-ms-mutability": ["create", "read"] attribute which is intended in the latest Stable API Version.

I have not updated these properties in any of the existing versions.

can you please help me out here on resolving this issue?

raosuhas commented:
Can you please get the approval from the breaking change board and then remove the "ARMChangesRequested" label when this PR is ready for review again .

ArcturusZhang commented on specification/datadog/resource-manager/readme.go.md:
````diff
@@ -21,10 +21,20 @@ azure-arm: true
 
 ``` yaml $(go) && $(multiapi)
 batch:
+  - tag: package-2022-06
   - tag: package-2021-03
   - tag: package-2020-02-preview
 ```
 
+### Tag: package-2022-06 and go
+
+These settings apply only when `--tag=package-2022-06 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2022-06' && $(go)
+output-folder: $(go-sdk-folder)/services/$(namespace)/mgmt/2022-06-01/$(namespace)
+```
+
````
We no longer support releasing more packages in go track 1 SDK any more since it is deprecating. Therefore could you please revert the changes in this file?

harsh007kumar replied:
@ArcturusZhang  does this mean for latest Api version we will not have GO SDK?

harsh007kumar committed:
reverted back changes from readme.go.md as these were used to serve track 1 SDK.

Now latest api-version will not have track 1 Go SDK, but it will have track 2 SDK.

ArcturusZhang commented:
The breaking changes reported in Go SDK come from recent changes in the code generator. Approved

harsh007kumar commented:
How do I remove a label? want to remove the 'no-recent-activity' label

tianxchen-ms commented:
Hi, @harsh007kumar breaking change approve is required.

harsh007kumar commented:
@raosuhas @tianxchen-ms I have got approval from the 'breaking change board' can you guys please review my PR now.
